common/sgx/collateral.c

```diff
@@ -194,6 +194,40 @@ static oe_result_t _parse_sgx_extensions(
     return result;
 }
 
+static oe_result_t _hex_crl_to_byte(
+    uint8_t* crl_hex,
+    size_t crl_hex_size,
+    uint8_t** crl_byte_buffer,
+    size_t* crl_byte_buffer_size)
+{
+    oe_result_t result = OE_OK;
+    size_t crl_byte_size = crl_hex_size >> 1;
+    uint8_t* crl_byte = (uint8_t*)oe_calloc(1, crl_byte_size);
+
+    if (crl_byte == NULL)
+        OE_RAISE(OE_OUT_OF_MEMORY);
+
+    for (size_t i = 0; i < crl_hex_size - 1; i++)
+    {
+        char c = (char)crl_hex[i];
+        int value = 0;
+        if (c >= '0' && c <= '9')
+            value = c - '0';
+        else if (c >= 'A' && c <= 'F')
+            value = 10 + c - 'A';
+        else if (c >= 'a' && c <= 'f')
+            value = 10 + c - 'a';
+        else
+            OE_RAISE_MSG(OE_INVALID_PARAMETER, "Invalid hex character: %c", c);
+        crl_byte[i >> 1] += value << (((i + 1) & 1) << 2);
```

Comment:
Can you explain a bit more on this line (e.g., adding comments)?

Reply:
Sure.